### PR TITLE
PV-106- Introduce PipelineOverview and related components.

### PR DESCRIPTION
**Pull request by jeder-rt**
Ref: https://pollywog.atlassian.net/browse/PV-106

### Issue
Need to bring in dynamic data for PipelineOverview

### Solution
- Added the PipelineOverview, PipelineSummary, PipelineMessage, & PipelineStageWrapper to hold the pipelineStage components. 
- Used border styles to created the angled slice shape on the pipeline view
- Upgraded to React 16.8 so we can use hooks!

### Testing Notes
- Created TestWrapper component to test components with theme styles

### QA Notes
QA Resource:
--none--

### Side Effects
- Also added react-spring (for future animations)
- PV-114, PV-115, PV-116, PV-117

### Dev Checklist
 - [x] There is a Jira Ticket for this changes
 - [x] The Jira ticket is assigned to me
 - [x] The Jira ticket is pointed
 - [x] The Jira Ticket is on the [board](https://pollywog.atlassian.net/secure/RapidBoard.jspa?projectKey=PV) and "In Code Review"
 - [x] I have run automated tests and manually tested my change prior to tagging others for code review
 - [x] I have run `make build` on the last commit (if applicable)
 - [x] All TODO comments have associated tickets

### Code Reviewer Checklist
- I have reviewed the code changes +1
- I have checked out the code and run it locally +10
- I have deployed the code and manually tested +100
- I have verified the Jira ticket associated with this PR is correct and has required fields

## Merging Checklist
- Do not not squash without checking
- Please delete the branch on merged
- Ensure Jira ticket is closed and in "Done" column on the [board](https://pollywog.atlassian.net/secure/RapidBoard.jspa?projectKey=PV)

Note: Pollywog PR template v0.1.0 - If making changes, please consider updating the other repositories as well

### CODE REVIEW

**Comment by jeder-rt** [comment]
Apologies for the massive PR. The dist files were updated and therefore became part of the PR. I think this is desired, but you could probably ignore these files on review. Please let me know if you want me to remove them.

File: pages/mocks/data/pipelineStats.json [hunk 1]
```diff
@@ -1,52 +1,265 @@
...
+        "amount": 10,
+        "unit": "days"
+      },
+      "projects_total": 10,
+      "projets_added": 2,
+      "projects_lost": 2,
+      "time_frame": "year"
+    }
+  ],
+  "period": [
```
Comment (bgarrett-rt) [inline pages/mocks/data/pipelineStats.json]: What's "period" again?

File: pages/mocks/pollywog1/Pipeline/PipelineMock.js [hunk 2]
```diff
@@ -7,8 +7,10 @@ import Typography from '@material-ui/core/Typography';
 import PipelineTableFilters from './Table/PipelineTableFilters';
 import PipelineTable from './Table/PipelineTableBody';
 
-import PipelineSummary from './Visual/PipelineSummary';
-import PipelineStages from './Visual/PipelineStages';
+// import PipelineSummary from './Visual/PipelineSummary';
```
Comment (bgarrett-rt) [inline pages/mocks/pollywog1/Pipeline/PipelineMock.js]: nit: Remove commented out imports

File: pages/mocks/pollywog1/Pipeline/PipelineMock.js [hunk 3]
```diff
@@ -7,8 +7,10 @@ import Typography from '@material-ui/core/Typography';
...
+import PipelineOverview from '../../../../src/modules/PipelineOverview';
+// import PipelineStages from './Visual/PipelineStages';
```
Comment (bgarrett-rt) [inline pages/mocks/pollywog1/Pipeline/PipelineMock.js]: nit: Remove commented out imports

File: pages/mocks/data/pipelineStats.json (hunk 1 above)
Comment (jeder-rt) [reply pages/mocks/data/pipelineStats.json]: period is what we select on (year, month, quarter, etc...)

File: src/modules/PipelineOverview/PipelineStageWrapper.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  });
+
+  return (
+    <div ref={containerRef}>
+      <TriangleOverlay width={containerWidth} height={30} inverted />
+      <Grid
+        container
+        alignItems={'center'}
+        direction={'row'}
+        // style={{ backgroundColor: 'red' }}
```
[the remark this answers is not in the document]
Comment (jeder-rt) [reply src/modules/PipelineOverview/PipelineStageWrapper.js]: Yeah, I plan on coming back to this, so I'll leave it here for future dev work.

**Comment by bgarrett-rt** [comment]
I approve, but at the moment, don't have the approval options through git. hrm.

**Pull request merged**